[wahkia] iP

wahkia opened the pull request.

Commit:
Add initial chatbot greeting and farewell code

This is the first entry of the project. It contains a Java program
that outputs a simple chatbot greeting and farewell message.

Commit:
Enhance chatbot with command echoing and custom responses

This commit adds functionality for echoing user commands and exiting
on 'bye'. The chatbot now features a customizable personality, with
random fun responses for unknown commands and special handling for
recognized commands like 'list'.

Commit:
Add task storage and listing functionality to Lia

This update enables Lia to store tasks entered by the user and
display them upon request with the 'list' command. The chatbot
now also acknowledges each added task and continues to offer
a personalized farewell when 'bye' is entered.

Commit:
Add task marking functionality and refactor to use Task class

This commit introduces the ability to mark tasks as done or not done
using the 'mark' and 'unmark' commands. A new Task class has been
added to represent tasks, encapsulating the task description and
status. The 'list' command now shows the status of each task.

Commit:
Refactor chatbot and Task class to follow coding standards

This commit refactors both the chatbot and Task class to comply
with Java coding standards. Javadoc comments have been added for
all methods and classes. The code layout, including indentation,
brace placement, and line length, has been adjusted.

Commit:
Add code for ToDo, Deadline, and Event tasks in Lia

This commit introduces three types of tasks—ToDo, Deadline, and Event—
to the Lia chatbot. Users can now add tasks with or without specific
dates and times. The tasks are displayed with their respective types
and details when listed. The code has been refactored to handle these
task types using inheritance, with each type represented by its own
class.

Commit:
Add more print line indents to better align ouput

Commit:
Refactor I/O redirection testing setup for Duke

Changed input.txt and updated the output
with the expected results in EXPECTED.TXT.

Commit:
Refactor task addition logic in Duke to reduce code duplication

This commit refactors the Duke chatbot by extracting the repetitive
task addition and confirmation printing logic into a new method
`addTaskAndPrint()`. This change improves code maintainability and
readability by reducing redundancy in handling "todo", "deadline",
and "event" commands.

kennethSty commented on src/main/java/Lia.java:
```diff
@@ -0,0 +1,126 @@
+
+    public static void main(String[] args) {
+        // Customizing the chatbot with the name Lia
+        String logo =
```
Coding Standard: Logo is constant. Use LOGO

kennethSty commented on src/main/java/Lia.java:
```diff
@@ -0,0 +1,126 @@
+
+        // Greet the user with enthusiasm
+        printLine();
+        System.out.println("    Hello! I'm \n" + logo);
+        System.out.println("    What can I do for you?");
```
Instead of leaving whitespaces use a constant String with the amount of whitespace and add them as INCIDENTION. This way UI bugs of not using enough whitespace can be avoided.

kennethSty commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,139 @@
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Returns a string representation of the Deadline task.
+     *
+     * @return A string representing the Deadline task.
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
+}
+
+/**
+ * Represents an Event task, which is a task that starts and ends
+ * at specific dates/times.
+ */
+class Event extends Task {
+    protected String start;
+    protected String end;
+
+    /**
+     * Creates a new Event task with the given description, start, and end times.
+     *
+     * @param description A description of the event.
+     * @param start The start time of the event.
+     * @param end The end time of the event.
+     */
+    public Event(String description, String start, String end) {
+        super(description);
+        this.start = start;
+        this.end = end;
+    }
+
+    /**
+     * Returns a string representation of the Event task.
+     *
```
Consider creating new files for each class.

kennethSty commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,11 @@
+list
+deadline return book /by Sunday
+list
+event project meeting /from Mon 2pm /to 4pm
```
consider adding  "wrong user input" to test whether the application breaks if e.g. deadline is used without /by.

kennethSty commented on src/main/java/Lia.java:
```diff
@@ -0,0 +1,126 @@
+        printLine();
+        System.out.println(" Got it. I've added this task:");
+        System.out.println("   " + task.toString());
+        System.out.println(" Now you have " + taskCount + " tasks in the list.");
```
Well done. The feedback to the user is helpful.

kennethSty reviewed (commented):
Your code is well-structured and functional. I’ve added a few comments about coding standards and quality improvements. Specifically, I suggest:

Indentation: Instead of manually inserting whitespace for formatting, consider defining an INDENTATION constant to maintain consistency and readability across your code.

Error Handling: To enhance system stability, especially when dealing with potential user input errors, consider wrapping your return values in an Optional object. This will provide a safer way to handle nulls and unexpected inputs.

Commit:
Refactor Lia chatbot code to adhere to coding standards

This commit introduces the following changes to comply with Java
coding standards:
- Added a constant 'INDENTATION' for consistent indentation in output.
- Updated the 'LOGO' variable to uppercase as a constant.
- Refactored the code layout for improved readability and maintainability.

Commit:
Refactor Task class into separate ToDo, Deadline, and Event subclasses

This commit restructures the Task class into three specific subclasses.
This change improves code organization and provides clearer modeling
of different task types.

Commit:
Add a new class called LiaException

Defined a class LiaException to represent exceptions specific to Lia.

Commit:
Add error handling to Lia chatbot using exceptions

This commit introduces custom error handling for the Lia chatbot
using a LiaException class. Errors such as missing descriptions for
todo tasks and invalid input for deadlines and events are now
handled with user-friendly error messages. Unrecognized commands
also trigger an error response.

Commit:
Merge branch 'branch-level-5'

Commit:
Refactor Task class to include ToDo, Deadline, and Event subclasses

This commit ensures that the task class and its subclasses can be packaged
in the future.

Commit:
Organize Lia chatbot into task and exception packages

This commit organizes the code into two packages:
- `task` package for task-related classes: Task, ToDo, Deadline, Event.
- `exception` package for Lia-specific exceptions: LiaException.

This improves code modularity and structure, making it easier to maintain and extend.

Commit:
Merge branch 'branch-A-Packages'

Commit:
Fix task count incrementation for non-task-adding commands

This commit refines how taskCount is handled in the Lia chatbot.
Previously, taskCount was incremented globally after each command,
leading to incorrect task counts for non-task-adding commands like
'list'. Now, taskCount is only incremented when tasks are actually
added ('todo', 'deadline', 'event'), preventing issues like
NullPointerExceptions during subsequent list commands.

Commit:
Reimplement task marking and unmarking functionality

This commit updates support for marking and unmarking tasks in the Lia
chatbot with error handling.

Commit:
Refactor Lia chatbot to use ArrayList for task management

This commit replaces the static Task[] array with an ArrayList<Task>
to dynamically handle task creation and management. This change
enhances flexibility in task operations like adding, listing, and
marking tasks, and ensures scalability without the limitations of a
fixed-size array.

Commit:
Add delete function to Lia chatbot

This commit introduces support for deleting tasks from the task list.
Users can now enter 'delete <task number>' to remove a specific task.
The remaining tasks are updated, and the user is notified of the
task removal and the current number of tasks.

Commit:
Add file saving and loading functionality to Lia chatbot

This commit introduces file persistence to the Lia chatbot, enabling
tasks to be saved to and loaded from a file (./data/lia.txt). Tasks
are automatically saved whenever the task list is modified and loaded
at startup. The system also handles missing files or directories.

Commit:
Add error handling for corrupted data in file loading

This commit enhances the Lia chatbot's ability to handle corrupted or
invalid data in the task file. The system now detects wrong task
entries, skips them, and prints warnings while continuing to load valid
tasks from the file.

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Lia.java

okkhoy commented on src/main/java/Lia.java:
```diff
@@ -0,0 +1,316 @@
+
+    public static void main(String[] args) {
+        // Customizing the chatbot with the name Lia
+        String LOGO =
```
make it constant?

okkhoy commented on src/main/java/Lia.java:
```diff
@@ -0,0 +1,316 @@
+        String input;
+        ArrayList<Task> tasks = loadTasks();  // Load tasks from file
+
+        // Greet the user with enthusiasm
```
this looks like a note-to-self comment or a redundant comment; can be removed

okkhoy commented on src/main/java/Lia.java:
```diff
@@ -0,0 +1,316 @@
+                saveTasks(tasks);
+
+            } catch (LiaException e) {
+                // Handle any Lia-specific exceptions
```
misleading comment since you are not doing anything specific apart from printing the message.

okkhoy commented on src/main/java/Lia.java:
```diff
@@ -0,0 +1,316 @@
+        Path filePath = Paths.get(FILE_PATH);
+
+        try {
+            if (!Files.exists(filePath)) {
```
make this as a guard clause;
eliminate the following arrow-head code

okkhoy reviewed (commented):
There are quite a few opportunities to eliminate arrow-head code. Try to use guard clauses to remove these.

Commit:
Refactor code to improve OOP structure

- Extracted Ui, Storage, Parser, and TaskList classes for better organization.
- Implemented Command pattern with specific command classes (AddCommand, DeleteCommand, etc.).
- Enhanced maintainability and readability of the codebase.

Commit:
Enhance chatBot to support date and time handling

Updated load method to parse dates using LocalDateTime.
Added a parseDateTime method for converting date strings to LocalDateTime objects.
Modified save method to ensure task dates are stored in the new format.
Ensured proper error handling for invalid date formats during loading.

wahkia committed:
Merge pull request #1 from wahkia/branch-level-8

Enhance chatBot to support date and time handling

Commit:
Implement find command to search for tasks by keyword

Added FindCommand class to handle task searching.
Updated Parser class to recognize the find command.
Integrated find functionality into the existing command execution flow.
Enhanced UI to display matching tasks to the user.

wahkia committed:
Merge pull request #2 from wahkia/branch-Level-9

Implement find command to search for tasks by keyword

Commit:
Add Javadoc comments tfor improved code documentation

wahkia committed:
Merge pull request #3 from wahkia/branch-A-JavaDoc

Add Javadoc comments for improved code documentation

Commit:
Add User Guide for Lia chatbot with usage instructions and features

wahkia committed:
Merge pull request #4 from wahkia/branch-A-UserGuide

Add User Guide for Lia chatbot with usage instructions and features

Commit:
Update on the style and design of the user guie

Commit:
Enhance Lia's personality in chatbot responses and add emojis for improved user engagement

Commit:
Upadate lia chatBot to not include emojis

Commit:
Add getting started section to the user guide

Commit:
Refactor the run command for Lia.jar
